janitor/timeseries.py

```diff
@@ -0,0 +1,123 @@
+"""
+Time series-specific data cleaning functions.
+"""
+
+import pandas as pd
+import pandas_flavor as pf
+from pandas.api.types import is_datetime64_any_dtype as is_datetime
+from collections import namedtuple
+from janitor import check
+
+
+def _flag_missing_timestamps(
+    df: pd.DataFrame,
+    frequency: str,
+    column_name: str,
+    first_time_stamp: pd.Timestamp,
+    last_time_stamp: pd.Timestamp,
+) -> namedtuple:
+    """
+    Utility function to test if input data frame is missing any timestamps relative to expected timestamps
+    generated based on the first_time_stamp, last_time_stamp and frequency.
+
+    :param pd.DataFrame df: data frame which needs to be tested for missing timestamps
+    :param str frequency: frequency i.e. sampling frequency of the data, expressed in seconds. A list of acceptable
+    frequency strings are available here
+    (https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases)
+    :param str column_name: name of the column which has time series if not the index.
+    :param pd.Timestamp first_time_stamp: timestamp at which the time_series is expected to start from.
+    :param pd.Timestamp last_time_stamp: timestamp at which the time_series is expected to end with.
```

Comment:
Types not needed, as they are in the function signature already.

```suggestion
    :param df: data frame which needs to be tested for missing timestamps
    :param frequency: frequency i.e. sampling frequency of the data, expressed in seconds. A list of acceptable
    frequency strings are available here
    (https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases)
    :param column_name: name of the column which has time series if not the index.
    :param first_time_stamp: timestamp at which the time_series is expected to start from.
    :param last_time_stamp: timestamp at which the time_series is expected to end with.
```